### PR TITLE
Provide strict k8s yaml model deserializer

### PR DESCRIPTION
**Pull request by stan-sz**

### CODE REVIEW

**Comment by stan-sz** [comment]
Fixes #1046

File: src/KubernetesClient.Models/KubernetesYaml.cs [hunk 1]
```diff
@@ -136,8 +142,9 @@ public static async Task<List<object>> LoadAllFromFileAsync(string fileName, IDi
         /// A map from apiVersion/kind to Type. For example "v1/Pod" -> typeof(V1Pod). If null, a default mapping will
         /// be used.
         /// </param>
+        /// <param name="strict">true if a strict deserializer should be used (throwing exception on unknown properties), false otherwise</param>
         /// <returns>collection of objects</returns>
-        public static List<object> LoadAllFromString(string content, IDictionary<string, Type> typeMap = null)
+        public static List<object> LoadAllFromString(string content, IDictionary<string, Type> typeMap = null, bool strict = false)
```
Comment (stan-sz) [inline src/KubernetesClient.Models/KubernetesYaml.cs:147]: TODO: add a unit test for strict LoadAllFromString

File: src/KubernetesClient.Models/KubernetesYaml.cs [hunk 2]
```diff
@@ -15,16 +15,22 @@ namespace k8s
...
-                .Build();
+                .WithOverridesFromJsonPropertyAttributes();
+        private static readonly IDeserializer StrictDeserializer =
+            CommonDeserializerBuilder
+            .Build();
+        private static readonly IDeserializer Deserializer =
+            CommonDeserializerBuilder
+            .IgnoreUnmatchedProperties()
+            .Build();
+        private static IDeserializer GetDeserializer(bool strict) => strict ? StrictDeserializer : Deserializer;
```
Comment (tg123) [inline src/KubernetesClient.Models/KubernetesYaml.cs:33]: still thinking about the better way to inject something `staticly`

Comment (stan-sz) [reply src/KubernetesClient.Models/KubernetesYaml.cs:33]: The problem is the mix of two design patterns - factory pattern of the Yaml (de)serializer with stateless static API call from K8s client...

**Review by tg123 — APPROVED** [review]
/LGTM

will revisit the static inject solution which would benefit automapper as well

**Pull request merged**